python/pyiceberg/transforms.py

```diff
@@ -644,8 +751,89 @@ def can_transform(self, _: IcebergType) -> bool:
     def result_type(self, source: IcebergType) -> IcebergType:
         return source
 
+    def project(self, name: str, pred: BoundPredicate) -> Optional[UnboundPredicate]:
+        return None
+
     def to_human_string(self, _: IcebergType, value: Optional[S]) -> str:
         return "null"
 
     def __repr__(self) -> str:
         return "VoidTransform()"
+
+
+M = TypeVar("M")
+
+
+def _truncate_number(
+    name: str, pred: BoundLiteralPredicate, transform: Callable[[Optional[M]], Optional[M]]
+) -> Optional[UnboundPredicate]:
+    boundary = pred.literal
+
+    if not isinstance(boundary, (LongLiteral, DecimalLiteral, DateLiteral, TimeLiteral, TimestampLiteral)):
```

Comment:
I think we should remove `TimeLiteral` from here.

Reply:
Did this together with removing `.increment()` and `.decrement()` from the Literal 👍🏻